### PR TITLE
refactor porter

### PR DESCRIPTION
**Pull request by duanjiong**
1. Eip Address Management via CRD
2. refactor Porter High Availability
3. Changes to the BgpConf/BgpPeer API to be compatible with the gobgp API and to support viewing status.
4. Some other changes

Signed-off-by: Duan Jiong <djduanjiong@gmail.com>

### CODE REVIEW

File: config/samples/network_v1alpha2_bgpconf.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: network.kubesphere.io/v1alpha2
+kind: BgpConf
+metadata:
+  #The porter only recognizes configurations with default names;
+  #configurations with other names are ignored.
+  name: default
```
Comment (zheng1) [inline config/samples/network_v1alpha2_bgpconf.yaml:6]: How about set the name to `porter`

Comment (duanjiong) [reply config/samples/network_v1alpha2_bgpconf.yaml:6]: This is fine, but the name `default` is used both in the code and in the documentation, so I don't want to change it, and it's not really perceptible to the user when it's configured through the UI.

**Comment by zheng1** [comment]
Here is some error
```
root@master:~# kubectl apply -f https://raw.githubusercontent.com/kubesphere/porter/master/deploy/porter.yaml
namespace/porter-system unchanged
customresourcedefinition.apiextensions.k8s.io/bgpconfs.network.kubesphere.io configured
customresourcedefinition.apiextensions.k8s.io/bgppeers.network.kubesphere.io configured
customresourcedefinition.apiextensions.k8s.io/eips.network.kubesphere.io configured
role.rbac.authorization.k8s.io/leader-election-role unchanged
error: error parsing https://raw.githubusercontent.com/kubesphere/porter/master/deploy/porter.yaml: error converting YAML to JSON: yaml: line 44: did not find expected '-' indicator
```

**Comment by duanjiong** [comment]
/assign @FeynmanZhou   

> Here is some error
> 
> ```
> root@master:~# kubectl apply -f https://raw.githubusercontent.com/kubesphere/porter/master/deploy/porter.yaml
> namespace/porter-system unchanged
> customresourcedefinition.apiextensions.k8s.io/bgpconfs.network.kubesphere.io configured
> customresourcedefinition.apiextensions.k8s.io/bgppeers.network.kubesphere.io configured
> customresourcedefinition.apiextensions.k8s.io/eips.network.kubesphere.io configured
> role.rbac.authorization.k8s.io/leader-election-role unchanged
> error: error parsing https://raw.githubusercontent.com/kubesphere/porter/master/deploy/porter.yaml: error converting YAML to JSON: yaml: line 44: did not find expected '-' indicator
> ```

There was an indentation error, the keyboard mishandled when I submitted it.

**Comment by FeynmanZhou** [comment]
@Patrick-LuoYu Do we have the relevant guides for these [two features](https://github.com/kubesphere/porterlb/releases/tag/v0.4.0) in v0.4.0? How to test them from the end user side?

**Comment by duanjiong** [comment]
> @Patrick-LuoYu @duanjiong @zheng1 Do we have the relevant guides for these [two features](https://github.com/kubesphere/porterlb/releases/tag/v0.4.0) in v0.4.0? How to test them from the end user side?

End-users look at the documentation on the line, unclear we improve the documentation, there are problems with the issue we fix on the line, with the use of documentation is also a test, why do we have to get a separate set of test cases.

**Pull request merged**